### DISCUSSION TITLE
Should we use oboe::ManagedStream or std::shared_ptr<oboe::AudioStream> to implement streams?

### DISCUSSION THREAD

**Discussion by mjbattistello** [google/oboe#1558] [Q&A] [answered]
For quite some time we’ve been receiving some crash reports that seem to be related to the following [issue](https://github.com/google/oboe/issues/1126#issuecomment-739014632).
As @philburk indicates there, we should use a shared stream (std::shared_ptr<oboe::AudioStream>) instead of an oboe::ManagedStream. 
**Is that the way we should implement a stream nowadays?**
We’ve based our code on what’s indicated in the documentation [Adding Oboe to your project](https://github.com/google/oboe/blob/f89abeaa1030696082baff4fa9967b088f3e9961/docs/GettingStarted.md), and there oboe::ManagedStream is being used, so maybe the docs are not up to date, or it was left in that way for very simple cases as an introduction to Oboe.

**Comment by robertwu1** [comment, marked as answer]
[Here](https://github.com/google/oboe/blob/main/docs/GettingStarted.md)'s the current version of Adding Oboe to your project. You linked a really old commit.

The docs were updated in January 2021 with https://github.com/google/oboe/commit/d1376b0399d0ca1c389bee29d1afdd2dd21a6ceb

**Reply by mjbattistello** [reply]
Hi Robert!
You’re totally right!
Thank you very much!
